pkgs/development/ocaml-modules/ocamlformat-rpc-lib/default.nix

```diff
@@ -0,0 +1,23 @@
+{ lib, fetchurl, buildDunePackage, csexp, sexplib0 }:
+
+buildDunePackage rec {
+  pname = "ocamlformat-rpc-lib";
+  version = "0.19.0";
+
+  src = fetchurl {
+    url = "https://github.com/ocaml-ppx/ocamlformat/releases/download/${version}/ocamlformat-${version}.tbz";
+    sha256 = "sha256-YvxGqujwpKM85/jXcm1xCb/2Fepvy1DRSC8h0g7lD0Y=";
+  };
+
+  minimumOCamlVersion = "4.08";
+  useDune2 = true;
+
+  propagatedBuildInputs = [ csexp sexplib0 ];
+
+  meta = with lib; {
+    homepage = "https://github.com/ocaml-ppx/ocamlformat";
+    description = "Auto-formatter for OCaml code (RPC mode)";
+    license = licenses.mit;
+    maintainers = with maintainers; [ Zimmi48 marsam ];
```

Comment:
I used the same maintainers as for the ocamlformat tool itself. Hopefully @marsam and @Zimmi48 don't mind.

Reply:
I don't mind, even though I wasn't aware of the existence of `ocamlformat-rpc-lib`. And now that I am, I'm surprised that there is no dependency between `ocamlformat` and `ocamlformat-rpc-lib` one way or the other.

Another question that I have is: does the fact that `ocamlformat-rpc-lib` is packaged here only at version 0.19.0 means that users of `ocaml-lsp` do not get to choose which version of `ocamlformat` they use?

Reply:
>  And now that I am, I'm surprised that there is no dependency between ocamlformat and ocamlformat-rpc-lib one way or the other.

That's because the ocamlformat binary does not use the protocol to handle formatting request.

> is packaged here only at version 0.19.0 means that users of ocaml-lsp do not get to choose which version of ocamlformat they use?

`ocamlformat-rpc-lib` is just the library that defines the protocol. So users are only asked to use a version of `ocamlformat-rpc` (not packaged in nix yet) that agrees on the protocol in `ocamlformat-rpc-lib`. Note that if no such version is found, ocamllsp degrades gracefully and simply doesn't offer features that depend on ocamlformat-rpc.

Reply:
OK thanks for the explanations!